### PR TITLE
UX-1100/storage creation form designed

### PR DESCRIPTION
**Pull request by rajshekhar23**
Screenshot 2021-08-02 at 11 58 32 PM
Screenshot 2021-08-02 at 11 58 43 PM

### CODE REVIEW

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js [hunk 1]
```diff
@@ -82,60 +131,74 @@ const BasicStep = ({ onSubmit, triggerSubmit, wizardContext }) => {
...
-          id="isDefault"
-          label="Use as Default Storage Class"
-          info={defaultExists && 'A default storage class already exists on this cluster.'}
-          disabled={defaultExists}
-        />
-      </ValidatedForm>
+        <ValidatedForm
+          classes={{ root: classes.validatedFormContainer }}
+          onSubmit={onSubmit}
+          triggerSubmit={triggerSubmit}
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js:145]: pass `elevated: false` to remove the default card from the form

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js [hunk 2]
```diff
@@ -9,50 +9,99 @@ import PicklistField from 'core/components/validatedForm/PicklistField'
...
+        </>
+      )}
+    </Wizard>
+  </>
 )
 
 const BasicStep = ({ onSubmit, triggerSubmit, wizardContext }) => {
+  const classes = useStyles({})
+  const [statsVisible, setStatsVisble] = useState(true)
+  const toggleStats = useCallback(() => setStatsVisble(!statsVisible), [statsVisible])
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js:103]: This isn't even doing anything... did you read the ticket?

File: src/api-client/Qbert.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { CSIDriverResponse } from './../app/plugins/kubernetes/components/storage/csi-drivers/model'
```
Comment (jruddell) [inline src/api-client/Qbert.ts:1]: this is a really weird import path. We have alias paths to help with this

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js [hunk 4]
```diff
@@ -78,64 +148,201 @@ const BasicStep = ({ onSubmit, triggerSubmit, wizardContext }) => {
...
-    : ''
-
   return (
-    <WizardStep stepId="basic" label="Basic">
-      <ValidatedForm
-        onSubmit={onSubmit}
-        triggerSubmit={triggerSubmit}
-        title="Name Your Storage Class"
+    <WizardStep stepId="basic" label="Configure Storage Class">
+      <WizardMeta
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js:157]: WizardMeta is meant to be used as a parent. It should be below the wizard but above the wizard steps.

```
<Wizard>
  <WizardMeta>
    <WizardStep>
    </WizardStep>
    <WizardStep>
    </WizardStep>
  <WizardMeta>
<Wizard>
```

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js [hunk 5]
```diff
@@ -101,17 +101,24 @@ export const AddStorageClassForm = ({ onComplete }) => (
...
-            wizardContext={wizardContext}
-          />
-          <CustomizeStep
-            wizardContext={wizardContext}
-            onSubmit={setWizardContext}
-            triggerSubmit={onNext}
-          />
+          <WizardMeta
+            fields={wizardContext}
+            icon={<CloudProviderCard active type={CloudProviders.Aws} />}
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js:106]: why is the icon for this card always AWS? what is in the design?

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js [hunk 6]
```diff
@@ -154,195 +161,188 @@ const BasicStep = ({ onSubmit, triggerSubmit, wizardContext, setWizardContext })
...
+              {!statsVisible && (
+                <TextField
+                  id="provisioner"
+                  label="Select Provisioner"
+                  info="Select provisioner to use"
+                  required
                 />
-                {statsVisible && (
+              )}
+              {(wizardContext.cloudProviderType === 'aws' ||
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js:216]: I believe we have an enum that you should use here when checking cloud provider types

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js [hunk 7]
```diff
@@ -375,34 +375,27 @@ const CustomizeStep = ({ wizardContext, onSubmit, triggerSubmit }) => {
...
-        keyOverrides={wizardMetaFormattedNames}
-        calloutFields={wizardMetaCalloutFields}
+      <ValidatedForm
+        onSubmit={onSubmit}
+        triggerSubmit={triggerSubmit}
+        title="YAML"
+        link={
+          <>
+            <CancelButton className={classes.editButton}>EDIT</CancelButton>
+            <SearchBar className={classes.search} onSearchChange={() => {}} searchTerm={''} />
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js:385]: what is this doing?

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js [hunk 8]
```diff
@@ -154,195 +161,188 @@ const BasicStep = ({ onSubmit, triggerSubmit, wizardContext, setWizardContext })
...
-                  label="Use as Default Storage Class"
-                  info={defaultExists && 'A default storage class already exists on this cluster.'}
-                  disabled={defaultExists}
-                />
-              </FormFieldCard>
-            </>
-          )}
-        </ValidatedForm>
-      </WizardMeta>
+                  {values.azureDiskType === 'azure-disk managed standard' && (
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js:256]: I really dont like this diskType comparison. Is there something more specific we can work off of to know when to render these sections?

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js (hunk 7 above)
Comment (rajshekhar23) [reply src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js:385]: In the ticket description nothing is mentioned about search field. It is added as placeholder.

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js (hunk 5 above)
Comment (rajshekhar23) [reply src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js]: updated and committed.

File: src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js (hunk 8 above)
Comment (rajshekhar23) [reply src/app/plugins/kubernetes/components/storage/AddStorageClassPage.js]: Updated

File: src/app/plugins/kubernetes/components/storage/csi-drivers/actions.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import Bugsnag from '@bugsnag/js'
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/storage/csi-drivers/actions.js:1]: new files should be typescript not javascript

File: src/app/plugins/kubernetes/datakeys.model.ts [hunk 10]
```diff
@@ -56,6 +56,7 @@ export interface IDataKeys {
   PrometheusServiceMonitors: Array<IGenericClusterizedResponse<any>> // no model for this yet
   PrometheusAlertManagers: Array<IGenericClusterizedResponse<any>> // no model for this yet
   StorageClasses: Array<IGenericResource<GetClusterClusterRoleBindingsItem>>
+  CSIDrivers: Array<IGenericResource<GetClusterClusterRoleBindingsItem>>
```
Comment (jruddell) [inline src/app/plugins/kubernetes/datakeys.model.ts:59]: GetClusterClusterRoleBindingsItem doesn't look like the right type here.

**Review by jruddell — APPROVED** [review]
there are some pending fixes in a future PR so lets get this merged and then pull in the pending fixes.

**Pull request merged**